File: API/Entities/Review.cs
namespace API.Entities;

public class Review
{
    public int Id { get; set; }
    public required string Title { get; set; }
    public required string Comment { get; set; }
    public int Rating { get; set; } // 1-5 stars
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }

    // Foreign Keys
    public required string ProductId { get; set; }
    public required string UserId { get; set; }

    // Navigation properties
    public Product Product { get; set; } = null!;
    public AppUser User { get; set; } = null!;
}

File: Client/src/types/review.ts
export type ReviewUser = {
	id: string;
	userName: string;
	email?: string;
	imageUrl?: string;
};

export type Review = {
	id: number;
	title: string;
	comment: string;
	rating: number;
	createdAt: string;
	updatedAt?: string;
	productId: string;
	userId: string;
	user: ReviewUser;
};

export type ReviewStats = {
	totalReviews: number;
	averageRating: number;
	ratingDistribution: {
		five: number; four: number; three: number; two: number; one: number;
	}
};
